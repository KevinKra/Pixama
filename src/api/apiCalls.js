import API_KEY from "./apikey";

export const fetchPopularMovies = async () => {
  let response = await fetch(
    `https://api.themoviedb.org/3/discover/movie?certification_country=US&api_key=${API_KEY}`
  );
  const movies = await response.json();
  return movies.results;
};

//This method is not hooked up anywhere at this point
//data must be an object with keys of name, email and password
export const fetchNewUser = (url, data)  => {
  return fetch(url, 
    {
      method: 'POST',
      body: JSON.stringify( data ),
      headers: {
        'Content-Type': 'application/json'
    }
  })
  .then(response=> {
    if(!response.ok) {
      throw new Error ("Email already exists", response.message)
    } else {
      return response.json()
    }
  })
}

//this method is being imported and called in LoginCard component
export const fetchUser = (url, data)  => {

  return fetch(url, 
    {
      method: 'POST',
      body: JSON.stringify( data ),
      headers: {
        'Content-Type': 'application/json'
    }
  })
  .then(response=> {
    if(!response.ok) {
      throw new Error ("Not a valid login",response.message)
    } else {
      return response.json()
    }
  })
}

//Method to add a movie to a user's favorites, this method not hooked up to a component yet
// url in component method needs to be 'http://localhost:3000/api/users/favorites/new'
//data needs to be object like this: {movie_id: 550, user_id: 1, title: "Fight Club", poster_path: "/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg" , release_date: "1999-10-15", vote_average:8.4, overview: "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground \"fight clubs\" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion."}
//response will be the new favorite ID
export const postFavorite = (url, data)  => {
console.log(data)
return fetch(url, 
    {
      method: 'POST',
      body: JSON.stringify( data ),
      headers: {
        'Content-Type': 'application/json'
    }
  })
  .then(response=> {
    if(!response.ok) {
      throw new Error ("Add error",response.message)
    } else {
      return response.json()
    }
  })
};


export const fetchFavorites = (url)  => {

return fetch(url)
  .then(response=> {
    if(!response.ok) {
      throw Error ("Couldn't get favorites",response.message)
    } else {
      return response.json()
    }
  })
};