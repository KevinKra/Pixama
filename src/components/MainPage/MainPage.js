import React, { Component } from "react";
import "./MainPage.scss";
import Carousel from "../Carousel/Carousel";
import HeroImage from "../HeroImage/HeroImage";

export default class MainPage extends Component {
  render() {
    return (
      <section className="MainPage">
        <HeroImage />
        <section className="main-body">
          <Carousel title="Popular Movies" movies={this.props.movies} />
        </section>
      </section>
    );
  }
}
