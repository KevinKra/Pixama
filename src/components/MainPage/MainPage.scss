.MainPage {
  border: 1px solid red;
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  min-width: 100vw;
}

.main-body {
  border: 1px solid blue;
  width: 90%;
  min-height: 100vh;
  transform: translateY(-50px);
}
