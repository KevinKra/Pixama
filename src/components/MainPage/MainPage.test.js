import { MainPage } from "./MainPage";
import React from "react";
import { shallow } from "enzyme";
import * as apiCalls from "../../api/apiCalls";

describe('MainPage', () => {
  let wrapper;
  let instance;
  let props = {
    popularMovies: [],
    romanceMovies: [],
    favorites: [{ title: "movie1" }, { title: "movie2" }],
    addPopularMovies: jest.fn(),
    addRomanceMovies: jest.fn(),
    updateFavorites: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<MainPage {...props} />);
    instance = wrapper.instance();
  });

  it ('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call fetchMovies twice on mount', async () => {
    // instance.componentDidMount();
    const popMovies = await apiCalls.fetchMovies('');
    const romMovies = await apiCalls.fetchMovies('');
    expect(apiCalls.fetchMovies).toHavevBeenCalledTimes(2);
  });
})
