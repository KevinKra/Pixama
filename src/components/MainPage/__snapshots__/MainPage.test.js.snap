// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should match the snapshot 1`] = `
<section
  className="MainPage"
>
  <HeroImage />
  <section
    className="main-body"
  >
    <Connect(Carousel)
      genre="favorites"
      title="Favorites"
    />
    <Connect(Carousel)
      genre="popularMovies"
      title="Popular Movies"
    />
    <Connect(Carousel)
      genre="romanceMovies"
      title="Romance Movies"
    />
  </section>
</section>
`;
